megamek/src/megamek/client/bot/princess/BehaviorSettingsFactory.java

```diff
@@ -229,7 +244,7 @@ public boolean saveBehaviorSettings(boolean includeTargets) {
         init(false);
         List<String> names;
         synchronized (behaviorMap) {
-            names = new ArrayList<String>(behaviorMap.keySet());
+            names = new ArrayList<>(behaviorMap.keySet());

```

Comment:
Looks like a bad change here that has nothing to do with log4j. I'm not sure where this came from, but this should definitely not have lost it's type information.